### PR TITLE
feat(aws_s3 sink): Add several S3 object options

### LINKED ISSUE #1686: Map Rusoto S3 options to the `aws_s3` sink
Following up on https://github.com/timberio/vector/issues/1673#issuecomment-581531537.

We should map the following [Rusoto options](https://rusoto.github.io/rusoto/rusoto_s3/struct.PutObjectRequest.html) to our own S3 sink. This allows users better control over ACLs, storage classes, etc.

- [ ] `acl`
- [ ] `grant_*`
- [ ] `server_side_encryption`
- [ ] `ssekms_key_id`
- [ ] `storage_class`
- [ ] `tags` (should allow field interpolation, ex: `{{field_name}}`

It probably makes sense to make these changes in multiple pull requests since we'll want to couple documentation changes with them.

### PR DESCRIPTION
**Pull request by bruceg**
Close #1686 

Signed-off-by: Bruce Guenter <bruce@timber.io>

### CODE REVIEW

**Comment by binarylogic** [comment]
Nice work! Although, I think some of these new options could be `enum`s, correct?

**Comment by bruceg** [comment]
Ah yes, a couple of them. I'll fix that.

File: .meta/sinks/aws_s3.toml [hunk 1]
```diff
@@ -75,3 +75,80 @@ examples = [
...
+common = false
+required = false
+description = "Gives the named grantee READ, READ_ACP, and WRITE_ACP permissions on the created objects."
+
+[sinks.aws_s3.options.grant_read]
+type = "string"
+category = "Object Attributes"
+common = false
+required = false
+description = "Allows the named grantee to read the created objects and their metadata."
```
Comment (LucioFranco) [inline .meta/sinks/aws_s3.toml]: Its not completely clear what needs to be set in this field? Maybe an example might be helpful?

Comment (bruceg) [reply .meta/sinks/aws_s3.toml]: When creating an object, you can name another entity to have read, write, or full access to the object. This names that other entity. FWIW the wording is almost straight out of the S3 put object docs.

**Comment by bruceg** [comment]
@binarylogic Does those enum additions resolve your question?

**Comment by LucioFranco** [comment]
@bruceg yeah that is much better thanks :+1:

**Pull request merged**